src/ess/reflectometry/gui.py

```diff
@@ -367,10 +458,393 @@ def log_progress(self, progress):
+            # Get the current plot
+            current_plot = plot_output.current_plot
+            # Get the figure's artists
+            artists = current_plot.artists
+
+            # Apply or revert transformation for each artist
+            for name, artist in artists.items():
+                if change['new']:  # Button is pressed - apply transformation
+                    # Store original data if not already stored
+                    if name not in original_data:
+                        original_data[name] = artist._data.copy()
+
+                    # Get x and y data from the artist
+                    x_data = artist._coord.values
+                    y_data = artist._data
+
+                    # Make sure x_data matches y_data shape
+                    if x_data.shape != y_data.values.shape:
+                        x_centers = (x_data[1:] + x_data[:-1]) / 2
+                        new_values = y_data.copy()
+                        valid_mask = ~np.isnan(y_data.values)
+                        transformed_values = y_data.values * x_centers**4
+                        new_values.values = np.where(
+                            valid_mask, transformed_values, y_data.values
+                        )
+                        if y_data.variances is not None:
+                            new_values.variances = np.where(
+                                valid_mask,
+                                y_data.variances * x_centers**8,
+                                y_data.variances,
+                            )
+                    else:
+                        new_values = y_data.copy()
+                        valid_mask = ~np.isnan(y_data.values)
+                        transformed_values = y_data.values * x_data**4
+                        new_values.values = np.where(
+                            valid_mask, transformed_values, y_data.values
+                        )
+                        if y_data.variances is not None:
+                            new_values.variances = np.where(
+                                valid_mask,
+                                y_data.variances * x_data**8,
+                                y_data.variances,
+                            )
+                else:  # Button is unpressed - revert to original
+                    new_values = original_data[name].copy()
+
+                # Update the line with new values
+                artist.update(new_values)
+
+            # Update view limits to show all data
+            fix_axis_orientation(current_plot, is_transformed=change['new'])
+
+            # Update without autoscaling
+            current_plot.update()
+
+        # Now override the methods
+        plot.view.canvas.reset_mode = custom_reset_mode
+        plot.view.canvas.zoom = custom_zoom
+        plot.view.canvas.panzoom = custom_panzoom
+        plot.view.canvas.draw = custom_draw
+
+        # Create and populate legend controls
+        def create_legend_controls():
```

Comment:
This seem like more GUI features have crept it, and keeping everything in sync is becoming more and more difficult.